client/handler.go

```diff
@@ -146,25 +149,76 @@ func (h *Handler) Update(w http.ResponseWriter, r *http.Request, ps httprouter.P
 		return
 	}
 
-	var secret string
-	if len(c.Secret) > 0 {
-		secret = c.Secret
+	c.OutfacingID = ps.ByName("id")
+	if err := h.updateClient(r.Context(), &c); err != nil {
+		h.r.Writer().WriteError(w, r, err)
+		return
 	}
 
-	c.OutfacingID = ps.ByName("id")
-	if err := h.r.ClientValidator().Validate(&c); err != nil {
+	h.r.Writer().Write(w, r, &c)
+}
+
+// swagger:route PATCH /clients/{id} admin patchOAuth2Client
+//
+// Patch an OAuth 2.0 Client
+//
+// Patch an existing OAuth 2.0 Client. If you pass `client_secret` the secret will be updated and returned via the API. This is the only time you will be able to retrieve the client secret, so write it down and keep it safe.
+//
+// OAuth 2.0 clients are used to perform OAuth 2.0 and OpenID Connect flows. Usually, OAuth 2.0 clients are generated for applications which want to consume your OAuth 2.0 or OpenID Connect capabilities. To manage ORY Hydra, you will need an OAuth 2.0 Client as well. Make sure that this endpoint is well protected and only callable by first-party components.
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: oAuth2Client
+//       500: genericError
+func (h *Handler) Patch(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	patchJSON, err := io.ReadAll(r.Body)
+	if err != nil {
 		h.r.Writer().WriteError(w, r, err)
 		return
 	}
 
-	c.UpdatedAt = time.Now().UTC().Round(time.Second)
-	if err := h.r.ClientManager().UpdateClient(r.Context(), &c); err != nil {
+	id := ps.ByName("id")
+	c, err := h.r.ClientManager().GetConcreteClient(r.Context(), id)
+	if err != nil {
+		h.r.Writer().WriteError(w, r, err)
+		return
+	}
+
+	if err := x.ApplyJSONPatch(patchJSON, c); err != nil {
```

Comment:
It shouldn't be allowed to update the ID of the client as that is immutable iirc!

Reply:
Good catch, will fix

Reply:
fixed.